packages/order-experience-api/src/api/createController.ts

```diff
@@ -102,14 +102,18 @@ export class CreateController extends AbstractController<typeof requestSchema> {
     return this.created<any>(res, dto.serialize())
   }
 
-  private async savePaymentFilters(
+  private async saveOrderPaymentFilters(
     orderData: Order,
     paymentFilters: PaymentFilter[]
   ) {
-    paymentFilters.forEach((filter: any) => {
-      filter.referenceId = orderData.orderId
-      filter.referenceType = ReferenceType.ORDER
-    })
-    return await savePaymentFiltersAdmin(paymentFilters)
+    if (!!paymentFilters && paymentFilters.length > 0) {
+      paymentFilters.forEach((filter: any) => {
+        filter.referenceId = orderData.orderId
+        filter.referenceType = ReferenceType.ORDER
+      })
+      return await savePaymentFiltersAdmin(paymentFilters)
+    } else {
+      return undefined
```

Comment:
Could this default return be an empty array? I dont like to have functions that returns undefined.